graylog2-server/src/main/java/org/graylog2/configuration/IndexerDiscoveryProvider.java

```diff
@@ -104,6 +107,9 @@ public void onRetry(Attempt attempt) {
                                     } else {
                                         LOG.info("Datanode is not available. Retry #{}/{}", attempt.getAttemptNumber(), connectionAttempts);
                                     }
+
+                                    // let's try to provision certificates, maybe there are datanodes waiting for these
+                                    graylogCertificateProvisioner.runProvisioning();
```

Comment:
This needs to be moved out of the `if` clause. The URI will be available even though provisioning might be necessary

Reply:
You are right, thx, done.